walkbox.go

```diff
@@ -1,23 +1,30 @@
 package pctk
 
-import "log"
+import (
+	"log"
+	"math"
+
+	rl "github.com/gen2brain/raylib-go/raylib"
+)
 
 // Walkbox refers to a convex polygonal area that defines the walkable space for actors.
 type WalkBox struct {
 	walkBoxID string
 	enabled   bool
-	vertices  [4]*Positionf
+	vertices  [4]Positionf
+	scale     float32
 }
 
 // NewWalkBox creates a new WalkBox with the given ID and vertices.
 // It ensures the polygon formed by the vertices is convex. If not, it will cause a panic.
 // Why convex? Because you can draw a straight line/path between any two vertices inside the polygon
 // without needing to implement complex pathfinding algorithms.
-func NewWalkBox(id string, vertices [4]*Positionf) *WalkBox {
+func NewWalkBox(id string, vertices [4]Positionf, scale float32) *WalkBox {
```

Comment:
I think we should use `Position` here instead of `Positionf`, and then convert it into float variant inside the function. Take into account that the screen has no float coordinates, and for the implementer it would be easier to describe boxes using integer positions. In fact, the script engine is designed to support a integer-based position type, but not its floating point sibling.